xbmc/platform/darwin/osx/network/NetworkMacOS.mm

```diff
@@ -179,45 +181,47 @@
 std::vector<std::string> CNetworkMacOS::GetNameServers()
 {
   std::vector<std::string> result;
-
-  FILE* pipe = popen("scutil --dns | grep -F \"nameserver\" | uniq | tail -n2", "r");
-
-  if (pipe)
+  @autoreleasepool
   {
-    int fineNoPipe = ::fileno(pipe);
-
-    struct pollfd pollFd;
-    pollFd.fd = fineNoPipe;
-    pollFd.events = POLLIN;
-    pollFd.revents = 0;
+    // Create an SCPreferencesRef object for the system preferences
+    auto preferenceName = CFSTR("GetDNSInfo");
+    SCPreferencesRef preferences = SCPreferencesCreate(nullptr, preferenceName, nullptr);
+    if (!preferences)
+    {
+      CFRelease(preferenceName);
+      CLog::Log(LOGWARNING, "Unable to determine nameservers");
+      return result;
+    }
 
-    int pollResult = poll(&pollFd, 1, 1000); // Poll for 1 second
-    if (pollResult > 0 && (pollFd.revents & POLLIN))
+    // Get the current DNS configuration from system preferences
+    CFDictionaryRef currentDNSConfig =
+        static_cast<CFDictionaryRef>(SCPreferencesGetValue(preferences, kSCPrefNetworkServices));
+    if (currentDNSConfig && CFDictionaryGetCount(currentDNSConfig) > 0)
     {
-      char buffer[256] = {'\0'};
-      if (fgets(buffer, sizeof(buffer), pipe))
+      NSDictionary* networkServices = (__bridge NSDictionary*)currentDNSConfig;
+      for (NSString* serviceKey in networkServices)
       {
-        // result looks like this - > '  nameserver[0] : 192.168.1.1'
-        // 2 blank spaces + 13 in 'nameserver[0]' + blank + ':' + blank == 18 :)
-        const unsigned int nameserverLineSize = 18;
-        std::vector<std::string> tmpStr = StringUtils::Split(buffer, "\n");
-        for (const std::string& str : tmpStr)
+        NSDictionary* service = networkServices[serviceKey];
+        NSDictionary* dnsSettings = service[static_cast<NSString*>(kSCEntNetDNS)];
+        NSArray* dnsServers = dnsSettings[static_cast<NSString*>(kSCPropNetDNSServerAddresses)];
+
+        if (dnsServers && dnsServers.count > 0)
```

Comment:
this check isn't needed at all

Reply:
@kambala-decapitator you mean only the count right? Shouldn't we still keep the null check for `NSArray* dnsServers` ?

Reply:
no, both can be dropped

Reply:
calling methods on `nil` is safe (noop)

Reply:
Always learning :)
So we can also drop the check for `currentDNSConfig` above right?

Reply:
yeah, not sure on that one due to bridging, wouldn't hurt to keep there